Dungeoneer/src/com/interrupt/dungeoneer/gfx/GlRenderer.java

```diff
@@ -778,6 +780,69 @@ public void updateShaderAttributes() {
 		}
 	}
 
+	public void renderTextBatches() {
+		ShaderProgram activeProgram;
+
+		if (renderingForPicking) {
+			activeProgram = ShaderManager.getShaderManager().getCompiledShader("picking").shader;
+		} else {
+			activeProgram = smoothLighting;
+		}
+
+		textBatch.setShader(activeProgram);
+
+		try {
+			int pvMatrixLoc = activeProgram.getUniformLocation("u_projectionViewMatrix");
+
+			for (int i = 0; i < textToRender.size; i++) {
+				DrawableText dT = textToRender.get(i);
+
+				textBatch.begin();
+
+				Gdx.gl.glDepthMask(true); // SpriteBatch.begin() forces off depth writing, which we need for 3D space.
+
+				Vector3 position = dT.parentPosition.cpy().add(dT.drawOffset);
```

Comment:
Calling `cpy()` every frame for this and the color are going to cause hitches, those throw away Vector3 and Color objects are going to fill up the pool and cause GC to kick in. It's ugly, but what we've done in the past is use some previously defined worker vectors / colors when we need to do stuff like this.

Reply:
I intended to review these once the functionality is correctly set, but I hadn't realised that it had a strong enough impact on the GC. I'll address other instances of object creation (per-frame) as well.